torch/optim/nadam.py

```diff
@@ -104,10 +109,12 @@ def step(self, closure=None):
                   eps=group['eps'],
                   decoupled_weight_decay=group['decoupled_weight_decay'],
                   foreach=group['foreach'],
+                  capturable=group['capturable'],
```

Comment:
Having "capturable" ingrained into the groups will also add it to the state_dict for people who are checkpointing. Do we want that? It seems strange for this to be exposed in the state_dict => maybe we want to remove it when users request for the state_dict and only compile it if the optimizer they're loading into is compiled with PT2.

The alternate user story would be that we read the state_dict from the user to know whether we should be PT2 compiling the optimizers (makes less sense to me)